### PR TITLE
Fix creating cluster with default VPC NAT mode

### LINKED ISSUE #938: create cluster with default VPC NAT mode fails with a CloudFormation template validation error
**What happened?**
`eksctl create cluster` with the single VPC NAT mode failed with a CloudFormation template validation error.

**What you expected to happen?**
For `eksctl create cluster` to work without explicitly specifying a VPC NAT mode.

**How to reproduce it?**
Run `eksctl create cluster` without specifying a VPC NAT mode, either via a config file or as a CLI arg.

**Anything else we need to know?**
This bug exists on the latest master only and not in the latest release 0.1.38.

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
 version.Info{BuiltAt:"1561467883", GitCommit:"0.1.38-52-g257de2c8-dirty", GitTag:""}
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}

```

**Logs**
```bash
eksctl create cluster --name=test --node-type=t3.small --nodes=3  --region=us-east-1  --v 4
...
2019-06-25T17:55:43+05:30 [▶]  failed task: create cluster control plane "test" (will not run other sequential tasks)
2019-06-25T17:55:43+05:30 [ℹ]  1 error(s) occurred and cluster hasn't been created properly, you may wish to check CloudFormation console
2019-06-25T17:55:43+05:30 [ℹ]  to cleanup resources, run 'eksctl delete cluster --region=us-east-1 --name=test'
2019-06-25T17:55:43+05:30 [✖]  creating CloudFormation stack "eksctl-test-cluster": ValidationError: Template format error: Unresolved resource dependencies [PrivateRouteTableUSEAST1B] in the Resources block of the template
	status code: 400, request id: 59bdc587-9744-11e9-9b9d-cb4920904d52
2019-06-25T17:55:43+05:30 [✖]  failed to create cluster "test"

```

### PR DESCRIPTION
**Pull request by cPu1**
### Description

Closes #938 

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] Manually tested

### CODE REVIEW

**Comment by martina-if** [comment]
LGTM

File: pkg/cfn/builder/vpc.go [hunk 1]
```diff
@@ -128,7 +131,8 @@ func (c *ClusterResourceSet) addNATGateways() error {
 	case api.ClusterDisableNAT:
 		c.noNAT()
 	default:
-		return errors.New(*c.spec.VPC.NAT.Gateway + "is not a valid NAT gateway mode")
+		// TODO validate this before starting to add resources
```
Comment (errordeveloper) [inline pkg/cfn/builder/vpc.go]: Do you want to do it in this PR or separately? I'm okay with doing it separately, but better to open an issue. There are a few things of this nature that could be improved, we also had a more general issue that needs closing on - #488.

Comment (cPu1) [reply pkg/cfn/builder/vpc.go]: I was thinking of doing it in a separate PR. The issue you linked to should address this as well so I don't think we need to open an issue specifically for this.

File: pkg/cfn/builder/vpc.go [hunk 2]
```diff
@@ -128,7 +131,8 @@ func (c *ClusterResourceSet) addNATGateways() error {
...
+		return errors.New(*c.spec.VPC.NAT.Gateway + " is not a valid NAT gateway mode")
```
Comment (errordeveloper) [inline pkg/cfn/builder/vpc.go]: Would you mind changing this to `fmt.Errorf`? It's currently preferred for new errors, we used `errors` package only for wrapping.

File: pkg/ctl/create/cluster.go [hunk 3]
```diff
@@ -75,7 +75,7 @@ func createClusterCmd(rc *cmdutils.ResourceCmd) {
 			api.SubnetTopologyPublic:  fs.StringSlice("vpc-public-subnets", nil, "re-use public subnets of an existing VPC"),
 		}
 		fs.StringVar(&params.kopsClusterNameForVPC, "vpc-from-kops-cluster", "", "re-use VPC from a given kops cluster")
-		fs.StringVar(cfg.VPC.NAT.Gateway, "vpc-nat-mode", "single", "VPC NAT mode, valid options: HighlyAvailable, Single, Disable")
+		fs.StringVar(cfg.VPC.NAT.Gateway, "vpc-nat-mode", api.ClusterSingleNAT, "VPC NAT mode, valid options: HighlyAvailable, Single, Disable")
```
Comment (errordeveloper) [inline pkg/ctl/create/cluster.go]: Ah, I see, so we just mixed up casing on review! :)

File: pkg/cfn/builder/vpc.go (hunk 2 above)
Comment (cPu1) [reply pkg/cfn/builder/vpc.go]: Sure. BTW, this is from the original PR, I just added a whitespace.

**Comment by cristian-radu** [comment]
LGTM. More error handling = better :)

File: pkg/ctl/create/cluster.go (hunk 3 above)
Comment (cristian-radu) [reply pkg/ctl/create/cluster.go]: Looks like I missed one, my bad.

Comment (cPu1) [reply pkg/ctl/create/cluster.go]: It was very minor, it happens :)

**Pull request merged**